### PR TITLE
EVG-20594: Replace DefaultTaskActivator with more accurate names

### PR DESCRIPTION
**Pull request by hadjri**
EVG-20594

### Description

- Deprecating the `DefaultTaskActivator ` while keeping it as a valid system activator for backwards compatibility purposes
- Added a more specifcally named activator for all internal non-userID activations that were using the `DefaultTaskActivator` before
- Add an explicit requirement for user ID retrieval during patch scheduling to use for the activator name, which shouldn't be an issue since users must be signed in to do this

### Testing
Existing tests passing

### CODE REVIEW

File: globals.go [hunk 1]
```diff
@@ -213,9 +213,29 @@ const (
 	// app server should log to stdout.
 	standardOutputLoggingOverride = "STDOUT"
 
-	DefaultTaskActivator   = ""
-	StepbackTaskActivator  = "stepback"
+	// DefaultTaskActivator is a deprecated legacy activator that used
```
Comment (ablack12) [inline globals.go:216]: should we make a ticket so we don't forget to get rid of this at some point? Presumably we can drop it a year from now

File: globals.go [hunk 2]
```diff
@@ -213,9 +213,29 @@ const (
...
+	// to be a majority of non-stepback and non-API activations.
+	DefaultTaskActivator = ""
+	// APIServerTaskActivator represents Evergreen's internal API activator
 	APIServerTaskActivator = "apiserver"
+	// StepbackTaskActivator represents the activator for tasks activated
+	// due to stepback.
+	StepbackTaskActivator = "stepback"
```
Comment (ablack12) [inline globals.go:223]: would it be chaotic to have all of these say activator at the end? Or does that seem redundant for the use cases where we see these?

File: globals.go [hunk 3]
```diff
@@ -213,9 +213,29 @@ const (
 	// app server should log to stdout.
 	standardOutputLoggingOverride = "STDOUT"
 
-	DefaultTaskActivator   = ""
-	StepbackTaskActivator  = "stepback"
+	// DefaultTaskActivator is a deprecated legacy activator that used
+	// to be a majority of non-stepback and non-API activations.
+	DefaultTaskActivator = ""
```
Comment (ablack12) [inline globals.go:218]: can we change the name of this to LegacyTaskActivator so new code knows not to use it?

File: model/version_activation.go [hunk 4]
```diff
@@ -120,7 +120,7 @@ func ActivateElapsedBuildsAndTasks(v *Version) (bool, error) {
 	}
 	if len(buildIdsToActivate) > 0 {
 		// Don't need to set the version in here since we do it ourselves in a single update
-		if err := build.UpdateActivation(buildIdsToActivate, true, evergreen.DefaultTaskActivator); err != nil {
+		if err := build.UpdateActivation(buildIdsToActivate, true, evergreen.BuildActivator); err != nil {
```
Comment (ablack12) [inline model/version_activation.go:123]: I'm actually maybe confused about the difference between the activation here and below but is it confusing to have two different activator names here?

Comment (ablack12) [reply model/version_activation.go:123]: Oh it seems like we activate the build if either the whole build or individual tasks are being activated. Maybe for traceability it's fine to leave this as is.

File: globals.go (hunk 1 above)
Comment (hadjri) [reply globals.go:216]: Created https://jira.mongodb.org/browse/EVG-20869

File: globals.go (hunk 2 above)
Comment (hadjri) [reply globals.go]: I think it's a good suggestion, it'll help to be more explicit with our naming and make it clear that the only non-users that activate things in Evergreen must be "activators". Added

**Review by ablack12 — APPROVED** [review]
Sorry for the massive review delay!

**Pull request merged**